### PR TITLE
Keep only the 30 most recently used needle images

### PR DESCRIPTION
**Pull request by Martchus**
See https://progress.opensuse.org/issues/15438

### CODE REVIEW

File: backend/baseclass.pm [hunk 1]
```diff
@@ -849,10 +849,8 @@ sub set_tags_to_assert {
     my $mustmatch = $args->{mustmatch};
     my $timeout   = $args->{timeout} // $bmwqemu::default_timeout;
 
-    # free all needle images (https://progress.opensuse.org/issues/15438)
-    for my $n (needle->all()) {
-        $n->{img} = undef;
-    }
+    # keep only 30 most recently used images (https://progress.opensuse.org/issues/15438)
```
Comment (okurz) [inline backend/baseclass.pm]: magic number in comment?

File: t/01-test_needle.t [hunk 2]
```diff
@@ -337,6 +337,33 @@ is($res->{needle}->{name}, 'login_sddm.ref.perfect', "perfect match should win")
 $res = $img1->search([$imperfect, $workaround], 0.9, 0);
 is($res->{needle}->{name}, 'login_sddm.ref.workaround.imperfect', "workaround match should win");
 
+# test caching via needle->get_image
```
Comment (okurz) [inline t/01-test_needle.t]: because you add this header for this section, why not move it into a subtest?

File: backend/baseclass.pm (hunk 1 above)
Comment (Martchus) [reply backend/baseclass.pm]: I'll remove that number.

File: t/01-test_needle.t (hunk 2 above)
Comment (okurz) [reply t/01-test_needle.t]: ok, will you?

Comment (Martchus) [reply t/01-test_needle.t]: Not as part of this PR. The benefit wouldn't be that high in my opinion, e.g. the different sub tests would actually still be intertwined.

**Pull request merged**